### PR TITLE
MA-2435: Migrate DB update logic from Sails API (AwsAccountController.js) into StepFunction

### PR DESCRIPTION
**Pull request by chandramishra**
Purpose: Migrate DB update logic from Sails API (AwsAccountController.js) into StepFunction

For more details Please refer this link : https://mountdiablo.jira.com/browse/MA-2435

Changes: API & AWS GIT HUB MicroServices

‌

Unit Testing

NA

ES Lint:

Implemented

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,9 +1,22 @@
 
 var fs = require('fs');
 var AWS = require('aws-sdk');
+//var mysql = require('mysql');
+var sleep = require('sleep');
```
Comment (SonalOjha) [inline src/index.js]: is sleep been used?

File: src/models/mcawsModels.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const Sequelize = require('sequelize');
```
Comment (SonalOjha) [inline src/models/mcawsModels.js]: Are we planning to use both waterline and sequelize models?

**Review by SonalOjha — APPROVED** [review]
Apart from some minor comments, the changes looks good to me.

File: src/index.js [hunk 3]
```diff
@@ -1,9 +1,22 @@
...
+const McawsModels = require('./models/mcawsModels.js');
 
 var baseHandler = require('aws-services-lib/lambda/base_handler.js')
 
+function getRulesPayload(rule){
+  return new Promise(function(resolve,reject) {
+    if(rule.owner == "CUSTOM_LAMBDA"){
```
Comment (RajkumarShivage) [inline src/index.js]: What if the owner attribute is not present?

File: src/index_update_db.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      callback(err);
+      return;
+    }
+    const mcawsDbObj = new McawsModels(
+      process.env.DB_USERNAME,
+      passwd.Plaintext.toString('ascii'),
+      process.env.DB_HOST,
+      'msaws'
+    );
+    mcawsDbObj.AwsAccount(resp1 => {
```
Comment (RajkumarShivage) [inline src/index_update_db.js]: change it to "resp"

File: src/models/mcawsModels.js (hunk 2 above)
Comment (RajkumarShivage) [inline src/models/mcawsModels.js]: before we decide on Sequelize, have we done evaluation on Sequelize? Is it documented?

**Review by bhargavapratik — APPROVED** [review]
LGTM

File: src/models/mcawsModels.js (hunk 2 above)
Comment (chandramishra) [reply src/models/mcawsModels.js]: its independent from sails-api and  sequelize  is more easy and popular than waterline

File: src/index.js (hunk 3 above)
Comment (chandramishra) [reply src/index.js]: as of now its not using anywhere

File: src/index_update_db.js (hunk 4 above)
Comment (chandramishra) [reply src/index_update_db.js]: done

File: src/index.js (hunk 1 above)
Comment (chandramishra) [reply src/index.js]: made changes

File: src/models/mcawsModels.js (hunk 2 above)
Comment (chandramishra) [reply src/models/mcawsModels.js]: no any document but  Sequelize is more easy to use .  and based on internet people are more emphasize to use Sequelize than waterLine. 
and also its independent of sails api code so i thought we can use this orm

Comment (SonalOjha) [reply src/models/mcawsModels.js]: This means we end up creating two different versions of models, one in waterline and other in sequelize. Why do we require sails independent code is what my question is?

File: src/index.js (hunk 3 above)
Comment (RajkumarShivage) [reply src/index.js]: do you mean getRulesPayload is not used anywhere?

File: src/models/mcawsModels.js (hunk 2 above)
Comment (RajkumarShivage) [reply src/models/mcawsModels.js]: we wanted get away from Sails framework and we have never used Sails framework for micro services development. Microservices are written in plain nodejs(no framework).

File: src/models/mcawsModels.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+function getDbOrmObj(user, passwd, host, dbname) {
```
Comment (RajkumarShivage) [inline src/models/mcawsModels.js]: Where are we handling the errors(e.g. wrong password/user or database connection issue)

File: template.yaml [hunk 6]
```diff
@@ -7,11 +7,95 @@ Parameters:
...
+    Type: String
+    Description: user name of mcaws db
+  DbPassword:
+    Type: String
+    NoEcho: true
+    Description: password of mcaws db
+  DbHost:
+    Type: String
+    Description: Host of mcaws db
+  MasterMgmAwsId:
```
Comment (RajkumarShivage) [inline template.yaml]: it should be MgmtAWSId.

File: template.yaml [hunk 7]
```diff
@@ -7,11 +7,95 @@ Parameters:
...
+    Type: String
+    Description: AWS Id of Master Management Account
```
Comment (RajkumarShivage) [inline template.yaml]: why is it mentioned as Master Management Account, it should "Management Account"

File: template.yaml [hunk 8]
```diff
@@ -7,11 +7,95 @@ Parameters:
...
+    Description: Aws Id of Datadog Aws Account
+  BillingApiKey:
+    Type: String
+    NoEcho: true
+    Description: Access key of Billing System
+  BillingSecretKey:
+    Type: String
+    NoEcho: true
+    Description: API key of Billing System
+  BillingServer :
```
Comment (RajkumarShivage) [inline template.yaml]: we should have said BillingEndpoint instead.

**Comment by bhargavapratik** [comment]
@chandramishra  Remove Conflict.

File: src/models/mcawsModels.js (hunk 2 above)
Comment (RajkumarShivage) [reply src/models/mcawsModels.js]: but we need an ORM, we cannot make direct SQL calls.

File: src/models/mcawsModels.js (hunk 5 above)
Comment (chandramishra) [reply src/models/mcawsModels.js]: handle in index_updatedDB.js

File: src/index.js (hunk 3 above)
Comment (chandramishra) [reply src/index.js]: ignore previous comment 
master account id require to set only with custom config rule

File: src/models/mcawsModels.js (hunk 5 above)
Comment (RajkumarShivage) [reply src/models/mcawsModels.js]: I do not see it. Can you please point me to the code?

File: src/index.js (hunk 3 above)
Comment (RajkumarShivage) [reply src/index.js]: For case of custom config rule, we need to handle it

File: template.yaml (hunk 6 above)
Comment (chandramishra) [reply template.yaml]: If I am going to change just name then i need to change a lot of places 
its required to again test with all scenario .its take more time

File: template.yaml (hunk 8 above)
Comment (chandramishra) [reply template.yaml]: If I am going to change just name then i need to change a lot of places 
its required to again test with all scenario

**Comment by RajkumarShivage** [comment]
resolve conflicts.

File: template.yaml (hunk 8 above)
Comment (RajkumarShivage) [reply template.yaml]: ok do not change it.

File: template.yaml (hunk 6 above)
Comment (RajkumarShivage) [reply template.yaml]: ok do not change it.

File: src/models/mcawsModels.js (hunk 5 above)
Comment (chandramishra) [reply src/models/mcawsModels.js]: index_update_db.js
line number 41 and 48

**Review by RajkumarShivage — APPROVED** [review]

File: src/index.js (hunk 3 above)
Comment (chandramishra) [reply src/index.js]: If role owner is not "CUSTOM_LAMBDA" then no need set any thing that's why i did not put any else part

**Pull request merged**